File: server/db/problem.go
package db

import "gorm.io/gorm"

type Problem struct {
	gorm.Model       `json:"-"`
	Title            string `json:"title"`
	TitleSlug        string `json:"titleSlug"`
	Content          string `json:"content"`
	Difficulty       string `json:"difficulty"`
	Likes            int    `json:"likes"`
	Dislikes         int    `json:"dislikes"`
	ExampleTestCases string `json:"exampleTestCases"`
	Hints            string `json:"hints"`
}

type CodeSnippet struct {
	gorm.Model `json:"-"`
	Lang       string `json:"lang"`
	LangSlug   string `json:"langSlug"`
	Code       string `json:"code"`
}

File: leechcode-fe-app/src/app/models/problem.ts
export class Problem {
    
    title:string; 
	titleSlug:string; 
	content:string;
	difficulty:string;
	likes:number;
	dislikes:number;
	exampleTestCases:string;
	hints:string;
    constructor(title:string,titleSlug:string,content:string,difficulty:string,likes:number,
        dislikes:number,
        exampleTestCases:string,
        hints:string,)
        {
            
            this.title = title;
            this.titleSlug = titleSlug;
            this.content = content;
            this.difficulty = difficulty;
            this.likes = likes;
            this.dislikes = dislikes;
            this.exampleTestCases = exampleTestCases;
            this.hints = hints;

    }

    
}